test/jdk/javax/crypto/Cipher/TestEmptyModePadding.java

```diff
@@ -37,20 +37,45 @@
 public class TestEmptyModePadding {
 
     public static void main(String[] args) throws Exception {
-        Provider provider = Security.getProvider(System.getProperty("test.provider.name", "SunJCE"));
+        Provider provider = Security.getProvider(
+                System.getProperty("test.provider.name", "SunJCE"));
 
-        test("AES", provider);
-        test("AES/ECB/PKCS5Padding", provider);
-        test("AES//PKCS5Padding", provider);        // Empty mode
-        test("AES/CBC/", provider);                 // Empty padding
-        test("AES/ /NoPadding", provider);          // Mode is a space
-        test("AES/CBC/ ", provider);                // Padding is a space
-        test("AES/ / ", provider);                  // Both mode and padding are spaces
-        test("AES//", provider);                    // Both mode and padding are missing
+        System.out.println("Testing against " + provider.getName());
 
+        String[] testTransformations = {
+            // transformations w/ only 1 component, i.e. algo
+            " ",
+            // transformations w/ only 2 components
+            "AES/", "AES/ ", "AES/CBC",
+            "PBEWithHmacSHA512/224AndAES_128/",
+            "PBEWithHmacSHA512/256AndAES_128/ ",
+            "PBEWithHmacSHA512/224AndAES_128/CBC",
+            // 3-component transformations w/ empty component(s)
+            "AES//", "AES/ /", "AES// ", "AES/ / ",
+            "AES/CBC/", "AES/CBC/ ",
+            "AES//PKCS5Padding", "AES/ /NoPadding",
+            "PBEWithHmacSHA512/224AndAES_128//",
+            "PBEWithHmacSHA512/224AndAES_128/ /",
+            "PBEWithHmacSHA512/224AndAES_128// ",
+            "PBEWithHmacSHA512/224AndAES_128/ / ",
+            "PBEWithHmacSHA512/256AndAES_128/CBC/",
+            "PBEWithHmacSHA512/256AndAES_128/CBC/ ",
+            "PBEWithHmacSHA512/256AndAES_128//PKCS5Padding",
+            "PBEWithHmacSHA512/256AndAES_128/ /PKCS5Padding",
+        };
+
+        for (String t : testTransformations) {
+            test(t, provider);
+        };
```

Comment:
nit, `;` after for loop

```suggestion
        }
```